### PR TITLE
cross-links: improved calculation

### PR DESCRIPTION
**Pull request by McBen**
By converting latlng into the cartesian coordination system we can simplify the whole
calculation.

- lesser trigonometric functions calls for a little more multiplications
- don't have to deal with antimeridian
- doesn't use a internal class 

In my personal test it was 2-5times faster then the old algorithm. But this surely depends on system and is hard to measure.

### CODE REVIEW

**Comment by johnd0e** [comment]
Looks great for me.
Just minor remark: consider https://github.com/IITC-CE/ingress-intel-total-conversion/wiki/Code-Style

- we are still following es 5.1, and avoiding features like const. See #47
- indent should be 2 spaces

**Review by johnd0e — APPROVED** [review]
LGFM

File: plugins/cross-links.js [hunk 1]
```diff
@@ -1,293 +1,268 @@
 // @author         mcben
 // @name           Cross links
 // @category       Draw
-// @version        1.2.0
+// @version        1.3.0
 // @description    Checks for existing links that cross planned links. Requires draw-tools plugin.
 
 
-window.plugin.crossLinks = function() {};
+window.plugin.crossLinks = function () { };
```
Comment (johnd0e) [inline plugins/cross-links.js:8]: What if we make all namespace references shorter?
```js
var crossLinks = window.plugin.window.plugin.crossLinks = function () {};
```

And then:
```js
crossLinks.greatCircleArcIntersect = function...
crossLinks.testPolyLine = function...
// etc
```
BTW, is it necessary to expose all this functions in namespace?
May be better left some of them private?

File: plugins/cross-links.js [hunk 2]
```diff
@@ -1,293 +1,268 @@
...
 
-window.plugin.crossLinks.greatCircleArcIntersect = function(a0,a1,b0,b1) {
-  // based on the formula at http://williams.best.vwh.net/avform.htm#Int
+/**
+ * greatCircleArcIntersect
+ */
+window.plugin.crossLinks.greatCircleArcIntersect = function (a0, a1, b0, b1) {
```
Comment (johnd0e) [inline plugins/cross-links.js:13]: If this (and others) function can be useful for 3rd-party plugins, then better to add short description of it's action and arguments.

File: plugins/cross-links.js [hunk 3]
```diff
@@ -1,293 +1,268 @@
...
-       opacity: 0.7,
-       weight: 5,
-       interactive: false,
-       dashArray: '8,8',
+  var poly = L.geodesicPolyline(link.getLatLngs(), {
+    color: '#d22',
+    opacity: 0.7,
+    weight: 5,
+    interactive: false,
+    dashArray: '8,8',
```
Comment (johnd0e) [inline plugins/cross-links.js:175]: Why not put line options into namespace, for easy customizing?

File: plugins/cross-links.js [hunk 4]
```diff
@@ -1,293 +1,268 @@
...
 }
 
-var setup = function() {
-    if (window.plugin.drawTools === undefined) {
-       alert("'Cross-Links' requires 'draw-tools'");
-       return;
-    }
+var setup = function () {
+  if (window.plugin.drawTools === undefined) {
+    alert('\'Cross-Links\' requires \'draw-tools\'');
```
Comment (johnd0e) [inline plugins/cross-links.js:248]:
```suggestion
  if (!window.plugin) {
    alert("'Cross-Links' requires 'draw-tools'");
```

File: plugins/cross-links.js [hunk 5]
```diff
@@ -1,293 +1,268 @@
...
+      window.plugin.crossLinks.testAllLinksAgainstLayer(e.layer);
+    } else {
+      // all other event types - assume anything could have been modified and re-check all links
+      window.plugin.crossLinks.checkAllLinks();
+    }
+  });
 
-    window.addHook('linkAdded', window.plugin.crossLinks.onLinkAdded);
-    window.addHook('mapDataRefreshEnd', window.plugin.crossLinks.onMapDataRefreshEnd);
+  window.addHook('linkAdded', window.plugin.crossLinks.onLinkAdded);
```
Comment (johnd0e) [inline plugins/cross-links.js:265]: Shouldn't `linkRemoved` be handled too?

**Comment by johnd0e** [comment]
@McBen 
We can merge this PR as is, without addressing upper comments.
Please consider them as general development discussion.

**Comment by McBen** [comment]
This PR is about "greatCircleArcIntersect".

The other changes where made to match your format rules as good as possible (and were done by my IDE)

File: plugins/cross-links.js (hunk 1 above)
Comment (McBen) [reply plugins/cross-links.js:8]: I don't like chained assignments

Comment (johnd0e) [reply plugins/cross-links.js:8]: Well, it's actually does not matter.
Consider this:
```js
var crossLinks = function () {};
window.plugin.window.plugin.crossLinks = crossLinks;
```

File: plugins/cross-links.js (hunk 4 above)
Comment (McBen) [reply plugins/cross-links.js:248]: Rule nr.8:  Avoid Negative Conditionals

**Comment by johnd0e** [comment]
@McBen 
It's true, and I have zero objections about it's merging.

Still, I'd like to continue discussion somewere.
So should I open separate issue, like _"Some ways to enhance iitc plugins code, on the example of `'cross-links'`"_?

**Review by modos189 — APPROVED** [review]
good job!

**Pull request merged**